cmd/main.go

```diff
@@ -197,31 +215,48 @@ func preRun(cmd *cobra.Command, args []string) {
 			log.Fatal().Msg(err.Error())
 		}
 	}()
+
+	return nil
 }
 
-func postRun(cmd *cobra.Command, args []string) {
+func postRun(cmd *cobra.Command, args []string) error {
 	channels.WaitGroup.Wait()
 
 	cfg := config.LoadConfig("2ms", Version)
 
 	// -------------------------------------
 	// Show Report
 	if report.TotalItemsScanned > 0 {
-		report.ShowReport(stdoutFormatVar, cfg)
+		if err := report.ShowReport(stdoutFormatVar, cfg); err != nil {
+			return err
+		}
+
 		if len(reportPathVar) > 0 {
 			err := report.WriteFile(reportPathVar, cfg)
 			if err != nil {
-				log.Error().Msgf("Failed to create report file with error: %s", err)
+				return fmt.Errorf("failed to create report file with error: %s", err)
 			}
 		}
```

Comment:
Where are you deciding if the exit code will be 1 or 0 for `TotalItemsScanned > 0`?

Reply:
at the Execute() function

'''go
if err := rootCmd.Execute(); err != nil {
return err
}

if report.TotalSecretsFound > 0 && ShowError("errors") {
return fmt.Errorf("")
}
'''